[mingyu-wan] iP

mingyu-wan opened the pull request:
# MYBOT Pro
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes/)

MYBOT Pro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 🖋️

And it is **FREE!**

Features:
- [X] Creating tasks (todo, deadlines, event)
- [X] Removing tasks
- [X] Storing tasks
- [X] Marking/Unmarking tasks
- [X] Removing tasks
- [X] Finding tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
        new Duke("src/main/MYBOT.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

mingyu-wan committed:
Level-1

mingyu-wan committed:
Level-1

mingyu-wan committed:
Level-2

mingyu-wan committed:
Level-3

mingyu-wan committed:
Level-4

mingyu-wan committed:
A-TextUiTesting

mingyu-wan committed:
Level-5

mingyu-wan committed:
Level-6

mingyu-wan committed:
A-TextUiTesting

mingyu-wan committed:
level-7

mingyu-wan committed:
Revert "level-7"

This reverts commit f996b75f1b599acecb412e980b90c2374008a8ab.

mingyu-wan committed:
level-7

mingyu-wan committed:
Merge branch Level 7

mingyu-wan committed:
Level-8 Date and Time for deadlines

mingyu-wan committed:
Level-8 Change to Time and Date for deadline

mingyu-wan committed:
Level-8 Date and Time changes for event

mingyu-wan committed:
Merge Branch Level-8

mingyu-wan committed:
A-MoreOOP

mingyu-wan committed:
Merge branch A-MoreOOP

mingyu-wan committed:
A-Packages

mingyu-wan committed:
Merge Add gradle support

mingyu-wan committed:
Merge branch 'master' of github.com:mingyu-wan/ip into A-MoreOOP

* 'master' of github.com:mingyu-wan/ip:
  Bump gradle and lib version
  Add Gradle support

mingyu-wan committed:
Merge A-MoreOOP

Some changes to implementation + gradle

mingyu-wan committed:
minor changes

mingyu-wan committed:
JUnit Test Cases

mingyu-wan committed:
Merge JUnit Test Cases

mingyu-wan committed:
Level-9

mingyu-wan committed:
A-JavaDoc

mingyu-wan committed:
Level-9

mingyu-wan committed:
Merge branch A-JavaDoc

mingyu-wan committed:
Merge branch 'master' into branch-A-Level-9

mingyu-wan committed:
Merge Branch A level 9

mingyu-wan committed:
minor changes to more OOP

mingyu-wan committed:
Changes to Level-9

mingyu-wan committed:
Merge branch 'branch-A-Level-9' of github.com:mingyu-wan/ip into branch-A-Level-9

* 'branch-A-Level-9' of github.com:mingyu-wan/ip:
  A-JavaDoc

# Conflicts:
#	src/main/java/duke/Ui.java

mingyu-wan committed:
A-CodingStandards

mingyu-wan committed:
Merge branch 'master' into branch-A-CodingStandard

mingyu-wan committed:
Merge Branch A-coding standard

mingyu-wan committed:
Merge Branch A-Level 9

aexolate commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+ * It can add, retrieve, remove tasks and keep track of the task count.
+ */
+public class TaskList {
+    private List<Task> task_List;
```
Perhaps the variable names could be modified to be in camelCase?

aexolate commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,199 @@
+        }
+        System.out.println("Unrecognised day format: " + day +
+                "\nPlease key in day in format Mon OR Monday " +
+                "\nThe task has been stored without a dueday, remove task and key in correct format to make changes \n");
```
Perhaps the String could be split at the comma to keep the line length shorter?

aexolate commented on src/main/java/duke/MYBotExceptions.java:
```diff
@@ -0,0 +1,37 @@
+package duke;
+public class MYBotExceptions extends Exception {
```
I like that there are many types of exceptions defined, and the names of the exceptions are self-explanatory.

fuyiqiao replied:
I agree! I think its especially nice that the exceptions created are easily accessible in the same class

aexolate commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+
+    public List<Task> getTask_List(){
```
Perhaps the method names could be written in camelCase as well?

aexolate commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+            new File(filePath);
+            tasks = new TaskList();
+        }
+        command = new Commands(ui,storage,tasks);
```
Perhaps you could add whitespace between the commas like this? `new Commands(ui, storage, tasks)`

aexolate commented:
Overall, I think the code was quite good and easily readable. The classes were nicely abstracted and used appropriately. There were minor coding standard violations, but they can be easily detected and fixed with a checkstyle. The Javadocs could be more comprehensive as they were missing from the exception class.

aexolate reviewed (commented):
Overall, I think the code was quite good and easily readable. The classes were nicely abstracted and used appropriately. There were minor coding standard violations, but they can be easily detected and fixed with a checkstyle. The Javadocs could be more comprehensive as they were missing from the exception class.

mingyu-wan committed:
GFMD

fuyiqiao commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,158 @@
+ */
+public class Commands {
+    private Ui ui;
+    private TaskList task_List;
```
should be `taskList` instead

fuyiqiao commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,158 @@
+     * @param input The description of the Deadline task.
+     * @param by The deadline of the task.
+     */
+    public void addDeadlineTask(String input, String by) {
```
`by` can be more informative, such as `dueBy`

fuyiqiao commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,158 @@
+     *
+     * @param task_number The index of the task to be marked.
+     */
+    public void markTasks(int task_number) {
```
should be `markTask` (singular)

fuyiqiao commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,158 @@
+     *
+     * @param task_number The index of the task to be marked.
+     */
+    public void unmarkTasks(int task_number){
```
`unmarkTask`

fuyiqiao commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,158 @@
+     *
+     * @param task_number The index of the task to be removed.
+     */
+    public void removeTasks(int task_number) {
```
`removeTask`

fuyiqiao commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,135 @@
+            if (input.equals("list")) {
+                command.listTasks(tasks);
+            } else if (input.startsWith("mark ")) {
+                int task_index = Integer.parseInt(input.substring(5));
```
`taskIndex`

mingyu-wan committed:
Level-10

mingyu-wan committed:
Merge branch 'master' into branch-Level-10

mingyu-wan committed:
Merge branch 'master' into branch-Level-10

mingyu-wan committed:
Merge branch Level-10

mingyu-wan committed:
Use GFMD

mingyu-wan committed:
Use GFMD

mingyu-wan committed:
Use GFMD

mingyu-wan committed:
Add assertions to code

Assertions are important because they help catch and identify critical
errors early in the development process, ensuring that the code behaves
as expected and preventing regression in later stages of development.

Add assertions to the Commands, Task, Todo, Event and Deadline classes
to check that values passed in are not null.

This is done to prevent when there are modifications to code, no null
classes are accidentally passed in, causing unnecessary errors.

mingyu-wan committed:
Merge branch-A-Assertions

Add assertions to code

mingyu-wan committed:
Improve Code Quality

Improving code quality is important because it enhances software
reliability, maintainability, and scalability while reducing the
likelihood of bugs and costly future fixes.

Rename class and variable names to obviously display what the code is
doing, Delete unused codes is some classes, Simplify complicated
expressions, Modify long expressions, Add code comments to codes that
look slightly more complicated.

This is done so that the codes are easier to read and understand.

mingyu-wan committed:
Merge branch-A-CodingQuality

mingyu-wan committed:
Add CI

mingyu-wan committed:
Merge branch-A-CI

mingyu-wan committed:
BCD-Extension C-Detectduplicates

Adding this extension is important because it improves the MYBOT system
and warns users if event has already been added.

Modify all addXXTask methods to check if there is a task of the same
name. Add methods and parameters to task list such that it
can temporarily store the task. Modify the commands so that if user
confirms that they want to add the item again, they need not retype the
task.

mingyu-wan committed:
A-Checkstyle

mingyu-wan committed:
Improving Code Quality

mingyu-wan committed:
Modified exceptions to each have a class on its own

mingyu-wan committed:
Merge pull request branch-Improvements

Modified exceptions to each have a class on its own

mingyu-wan committed:
A-BetterGUI

mingyu-wan committed:
User Guide

mingyu-wan committed:
Update README.md

mingyu-wan committed:
Bug fixes to incorrect date, day and time input

mingyu-wan committed:
Merge branch 'master' of github.com:mingyu-wan/ip
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mingyu-wan committed:
More comprehensive testing

mingyu-wan committed:
Merge branch-A-MoreTesting

mingyu-wan committed:
minor modifications to UG

mingyu-wan committed:
Modifications for release
